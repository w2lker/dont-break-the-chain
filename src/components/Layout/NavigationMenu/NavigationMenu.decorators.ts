import withStyles from '@material-ui/core/styles/withStyles';

import NavigationMenu from './NavigationMenu';
import navigationMenuStyles from './NavigationMenu.styles';

// @ts-ignore
const styledComponent = withStyles(navigationMenuStyles)(NavigationMenu);

export default styledComponent;
