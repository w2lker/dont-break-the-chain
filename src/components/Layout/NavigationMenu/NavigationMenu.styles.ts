const navigationMenuStyles = {
  root: {
    position: 'relative',
    zIndex: 1,
    boxShadow: '0px -4px 8px rgba(176, 190, 197, 0.24)',
  },
};

export default navigationMenuStyles;
