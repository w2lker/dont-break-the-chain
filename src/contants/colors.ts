import colorBlue from '@material-ui/core/colors/blue';
import colorBrown from '@material-ui/core/colors/brown';
import colorCyan from '@material-ui/core/colors/cyan';
import colorDeepPurple from '@material-ui/core/colors/deepPurple';
import colorGreen from '@material-ui/core/colors/green';
import colorIndigo from '@material-ui/core/colors/indigo';
import colorOrange from '@material-ui/core/colors/orange';
import colorPink from '@material-ui/core/colors/pink';
import colorPurple from '@material-ui/core/colors/purple';
import colorTeal from '@material-ui/core/colors/teal';

export type colorsKeys = 'pink' | 'purple' | 'deepPurple' | 'indigo' | 'blue' | 'cyan' | 'teal' | 'green' | 'orange' | 'brown';

export const colors: {[color: string]: any}  = {
  blue: colorBlue,
  brown: colorBrown,
  pink: colorPink,
  cyan: colorCyan,
  deepPurple: colorDeepPurple,
  indigo: colorIndigo,
  green: colorGreen,
  orange: colorOrange,
  purple: colorPurple,
  teal: colorTeal,
};
