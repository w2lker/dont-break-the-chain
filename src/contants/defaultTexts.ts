const defaultTexts = {
  navigationMenu: {
    chains: 'Chains',
    habits: 'Habits',
    profile: 'Profile',
  },
  habits: {
    header: {
      title: 'My Habits',
    },
  },
  profile: {
    header: {
      title: 'Profile',
    },
    stats: {
      total: {
        description: 'points received',
      },
      days: {
        subtitle: 'days',
        description: 'longest chain',
      },
    },
  },
};

export default defaultTexts;
