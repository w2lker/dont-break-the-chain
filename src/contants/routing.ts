export const routing = {
  default: '/',
  chains: '/chains',
  habits: '/habit',
  profile: '/account',
};
