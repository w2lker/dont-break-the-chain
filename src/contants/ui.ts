export const defaultIcons = {
  navigationMenu: {
    chains: 'layers',
    habits: 'flag-checkered',
    profile: 'account',
  },
};

// TODO: provide real placeholder image and attach link to it here
export const avatarPlaceholder = './placeholder.png';
