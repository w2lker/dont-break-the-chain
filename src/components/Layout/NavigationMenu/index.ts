import NavigationMenu from './NavigationMenu.decorators';

export default NavigationMenu;
