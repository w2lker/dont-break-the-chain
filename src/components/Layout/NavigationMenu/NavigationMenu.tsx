import React from 'react';

import { BottomNavigation } from '@material-ui/core';

import NavigationMenuItem from '../NavigationMenuItem';

import defaultTexts from '../../../contants/defaultTexts';
import { routing } from '../../../contants/routing';
import { defaultIcons } from '../../../contants/ui';

import { INavMenuItemProps } from '../NavigationMenuItem/NavigationMenuItem';

interface INavigationMenuProps {
  classes?: any;
}

interface INavConfig {
  chains: INavMenuItemProps;
  habits: INavMenuItemProps;
  profile: INavMenuItemProps;
}

const NavigationMenu: React.FC<INavigationMenuProps> = (props) => {
  const { classes } = props;
  const texts = defaultTexts.navigationMenu;
  const icons = defaultIcons.navigationMenu;

  // @ts-ignore
  const navConfig: INavConfig = {
    // @ts-ignore
    chains: {
      // @ts-ignore
      url: routing.chains,
      label: texts.chains,
      icon: icons.chains,
    },
    // @ts-ignore
    habits: {
      // @ts-ignore
      url: routing.habits,
      label: texts.habits,
      icon: icons.habits,
    },
    // @ts-ignore
    profile: {
      // @ts-ignore
      url: routing.profile,
      label: texts.profile,
      icon: icons.profile,
    },
  };

  // @ts-ignore
  const navigationActionsRender = Object.values(navConfig).map((config: INavMenuItemProps) => {
    return <NavigationMenuItem key={config.url} {...config} />;
  });

  return (
    <div className={classes.root}>
      <BottomNavigation>
        {navigationActionsRender}
      </BottomNavigation>
    </div>
  );

};

export default NavigationMenu;
