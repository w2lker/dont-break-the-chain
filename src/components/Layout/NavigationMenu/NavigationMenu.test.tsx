import React from 'react';

import { mount, render, shallow } from 'enzyme';
import cases from 'jest-in-case';
import { MemoryRouter } from 'react-router-dom';

import NavigationMenu from './NavigationMenu';
import NavigationMenuDecorated from './NavigationMenu.decorators';

import defaultTexts from '../../../contants/defaultTexts';
import { routing } from '../../../contants/routing';
import { defaultIcons } from '../../../contants/ui';
import navigationMenuStyles from './NavigationMenu.styles';

const classes = {
  root: 'navigation-menu-root',
};

const classesKeys = Object.keys(classes);

describe('NavigationMenu', () => {
  const props = {
    classes,
  };

  const wrappedComponent = mount(
    <MemoryRouter>
      <NavigationMenu {...props} />
    </MemoryRouter>,
  );

  it('match snapshot', () => {
    expect(wrappedComponent.debug()).toMatchSnapshot();
  });

  it('renders 3 menu items', () => {
    const menuItems = wrappedComponent.find('NavigationMenuItem');
    expect(menuItems.length).toBe(3);
  });

  const texts = defaultTexts.navigationMenu;
  const icons = defaultIcons.navigationMenu;

  const casesTitle = 'Testing rendering props of MenuItems';

  const casesTester = (opts: typeof casesData[0]) => {
    const item = wrappedComponent.find('NavigationMenuItem').at(opts.id);
    expect(item).toBeDefined();
    expect(item.props()).toMatchObject({
      // @ts-ignore
      url: routing[opts.def],
      // @ts-ignore
      label: texts[opts.def],
      // @ts-ignore
      icon: icons[opts.def],
    });
  };

  const casesData = [
    { name: 'Chains is rendered', id: 0, def: 'chains' },
    { name: 'Habits is rendered', id: 1, def: 'habits' },
    { name: 'Profile is rendered', id: 2, def: 'profile' },
  ];

  cases(casesTitle, casesTester, casesData);
});

describe('NavigationMenu Styles', () => {
  it('has root class', () => {
    expect(navigationMenuStyles.root).toBeDefined();
  });
});

describe('NavigationMenu Decorated', () => {
  const wrappedComponent = mount(
      <MemoryRouter>
        <NavigationMenuDecorated />
      </MemoryRouter>,
    );
  const component = wrappedComponent.find('NavigationMenu');
  it('has root object', () => {
    expect(component.length).toBe(1);
  });
    // @ts-ignore
  const assignedClasses = component.props().classes;
  it('has root class in classes map', () => {
    expect(assignedClasses.root).toBeDefined();
  });
});
