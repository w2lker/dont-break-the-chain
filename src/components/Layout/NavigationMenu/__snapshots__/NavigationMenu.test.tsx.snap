// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu match snapshot 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <NavigationMenu classes={{...}}>
      <div className=\\"navigation-menu-root\\">
        <WithStyles(ForwardRef(BottomNavigation))>
          <ForwardRef(BottomNavigation) classes={{...}}>
            <div className=\\"MuiBottomNavigation-root\\">
              <WithStyles(NavigationMenuItem) url=\\"/chains\\" label=\\"Chains\\" icon=\\"layers\\" classes={{...}} selected={false} showLabel={false} value={0} onChange={[undefined]}>
                <NavigationMenuItem classes={{...}} url=\\"/chains\\" label=\\"Chains\\" icon=\\"layers\\" selected={false} showLabel={false} value={0} onChange={[undefined]}>
                  <WithStyles(ForwardRef(BottomNavigationAction)) classes={{...}} component={[function]} to=\\"/chains\\" label=\\"Chains\\" icon={{...}} showLabel={true}>
                    <ForwardRef(BottomNavigationAction) classes={{...}} component={[function]} to=\\"/chains\\" label=\\"Chains\\" icon={{...}} showLabel={true}>
                      <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/chains\\">
                        <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/chains\\">
                          <DecoratedNavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/chains\\">
                            <NavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/chains\\" activeClassName=\\"Mui-selected\\">
                              <Link aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} to={{...}} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false}>
                                <LinkAnchor aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/chains\\" navigate={[Function: navigate]}>
                                  <a aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/chains\\" onClick={[Function: onClick]}>
                                    <span className=\\"MuiBottomNavigationAction-wrapper\\">
                                      <MaterialDesignIcon className=\\"NavigationMenuItem-icon-5\\" name=\\"layers\\">
                                        <span className=\\"material-icon mdi mdi-layers NavigationMenuItem-icon-5\\" style={[undefined]} />
                                      </MaterialDesignIcon>
                                      <span className=\\"MuiBottomNavigationAction-label NavigationMenuItem-label-4\\">
                                        Chains
                                      </span>
                                    </span>
                                    <NoSsr>
                                      <WithStyles(undefined) center={false}>
                                        <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                          <span className=\\"MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </ForwardRef(TouchRipple)>
                                      </WithStyles(undefined)>
                                    </NoSsr>
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </DecoratedNavLink>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(BottomNavigationAction)>
                  </WithStyles(ForwardRef(BottomNavigationAction))>
                </NavigationMenuItem>
              </WithStyles(NavigationMenuItem)>
              <WithStyles(NavigationMenuItem) url=\\"/habit\\" label=\\"Habits\\" icon=\\"flag-checkered\\" classes={{...}} selected={false} showLabel={false} value={1} onChange={[undefined]}>
                <NavigationMenuItem classes={{...}} url=\\"/habit\\" label=\\"Habits\\" icon=\\"flag-checkered\\" selected={false} showLabel={false} value={1} onChange={[undefined]}>
                  <WithStyles(ForwardRef(BottomNavigationAction)) classes={{...}} component={[function]} to=\\"/habit\\" label=\\"Habits\\" icon={{...}} showLabel={true}>
                    <ForwardRef(BottomNavigationAction) classes={{...}} component={[function]} to=\\"/habit\\" label=\\"Habits\\" icon={{...}} showLabel={true}>
                      <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/habit\\">
                        <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/habit\\">
                          <DecoratedNavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/habit\\">
                            <NavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/habit\\" activeClassName=\\"Mui-selected\\">
                              <Link aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} to={{...}} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false}>
                                <LinkAnchor aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/habit\\" navigate={[Function: navigate]}>
                                  <a aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/habit\\" onClick={[Function: onClick]}>
                                    <span className=\\"MuiBottomNavigationAction-wrapper\\">
                                      <MaterialDesignIcon className=\\"NavigationMenuItem-icon-5\\" name=\\"flag-checkered\\">
                                        <span className=\\"material-icon mdi mdi-flag-checkered NavigationMenuItem-icon-5\\" style={[undefined]} />
                                      </MaterialDesignIcon>
                                      <span className=\\"MuiBottomNavigationAction-label NavigationMenuItem-label-4\\">
                                        Habits
                                      </span>
                                    </span>
                                    <NoSsr>
                                      <WithStyles(undefined) center={false}>
                                        <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                          <span className=\\"MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </ForwardRef(TouchRipple)>
                                      </WithStyles(undefined)>
                                    </NoSsr>
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </DecoratedNavLink>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(BottomNavigationAction)>
                  </WithStyles(ForwardRef(BottomNavigationAction))>
                </NavigationMenuItem>
              </WithStyles(NavigationMenuItem)>
              <WithStyles(NavigationMenuItem) url=\\"/account\\" label=\\"Profile\\" icon=\\"account\\" classes={{...}} selected={false} showLabel={false} value={2} onChange={[undefined]}>
                <NavigationMenuItem classes={{...}} url=\\"/account\\" label=\\"Profile\\" icon=\\"account\\" selected={false} showLabel={false} value={2} onChange={[undefined]}>
                  <WithStyles(ForwardRef(BottomNavigationAction)) classes={{...}} component={[function]} to=\\"/account\\" label=\\"Profile\\" icon={{...}} showLabel={true}>
                    <ForwardRef(BottomNavigationAction) classes={{...}} component={[function]} to=\\"/account\\" label=\\"Profile\\" icon={{...}} showLabel={true}>
                      <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/account\\">
                        <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" focusRipple={true} onClick={[Function: handleChange]} component={[function]} to=\\"/account\\">
                          <DecoratedNavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/account\\">
                            <NavLink className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} to=\\"/account\\" activeClassName=\\"Mui-selected\\">
                              <Link aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} to={{...}} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false}>
                                <LinkAnchor aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onClick={[Function: handleChange]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/account\\" navigate={[Function: navigate]}>
                                  <a aria-current={{...}} className=\\"MuiButtonBase-root MuiBottomNavigationAction-root NavigationMenuItem-root-3\\" style={[undefined]} onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} role=\\"button\\" aria-disabled={false} href=\\"/account\\" onClick={[Function: onClick]}>
                                    <span className=\\"MuiBottomNavigationAction-wrapper\\">
                                      <MaterialDesignIcon className=\\"NavigationMenuItem-icon-5\\" name=\\"account\\">
                                        <span className=\\"material-icon mdi mdi-account NavigationMenuItem-icon-5\\" style={[undefined]} />
                                      </MaterialDesignIcon>
                                      <span className=\\"MuiBottomNavigationAction-label NavigationMenuItem-label-4\\">
                                        Profile
                                      </span>
                                    </span>
                                    <NoSsr>
                                      <WithStyles(undefined) center={false}>
                                        <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                          <span className=\\"MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </ForwardRef(TouchRipple)>
                                      </WithStyles(undefined)>
                                    </NoSsr>
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </DecoratedNavLink>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(BottomNavigationAction)>
                  </WithStyles(ForwardRef(BottomNavigationAction))>
                </NavigationMenuItem>
              </WithStyles(NavigationMenuItem)>
            </div>
          </ForwardRef(BottomNavigation)>
        </WithStyles(ForwardRef(BottomNavigation))>
      </div>
    </NavigationMenu>
  </Router>
</MemoryRouter>"
`;
